packages/core/src/errors-list.ts

```diff
@@ -0,0 +1,322 @@
+export const ERROR_PREFIX = "IGN";
+
+/**
+ * ErrorDescriptor is a type that describes an error.
+ * It's used to generate error codes and messages.
+ *
+ * @beta
+ */
+export interface ErrorDescriptor {
+  number: number;
+  // Message can use templates. See applyErrorMessageTemplate
+  message: string;
+}
```

Comment:
I think Hardhat has extra fields here, but I figured we start with the minimum and can always add more later.